builder/builder.go

```diff
@@ -90,48 +82,39 @@ func (b *Builder) RunAllBuildSteps(ctx context.Context, pipeline *graph.Pipeline
 		return err
 	}
 
-	for k, v := range pipeline.Dag.Nodes {
-		if k == graph.RootNodeID {
+	var deps []*models.ImageDependencies
+	for _, n := range pipeline.Dag.Nodes {
+		step := n.Value
+		log.Printf("Step ID %v marked as %v (elapsed time in seconds: %f)\n", step.ID, step.StepStatus, step.EndTime.Sub(step.StartTime).Seconds())
+
+		if !step.IsBuildStep() { // If the step isn't a build step, it won't have dependencies from the scanner.
```

Comment:
>IsBuildStep [](start = 11, length = 11)

Is it possible to just check ImageDependencies is nil or empty so you don't need to special handle build?

Reply:
Fixed